rfcs/0000-central-station.md

```diff
@@ -0,0 +1,507 @@
+this, as opposed to using their API. My hope is that if we move forward with this, that will
+motivate Swiftly to implement it. And with the MBTA producing these for all its modes and Swiftly
+consuming them that might be the push to finally get Service Changes officially accepted.
+
+## RTR
+
+### Current description
+
+RTR provides the realtime data for the heavy rail and light rail routes (together: "subway"). It
+receives data from the OCC via Kinesis events. The events are for the following data:
+
+- Light rail (Green Line and Mattapan) train movements
+  - Track circuits mostly downtown
+  - GPS from the onboard rocket routers, matched into "geofences" which were manually created
+  - AVIs that the trains pass.
+- Heavy rail (Red, Orange, and Blue)
+  - Track circuits only
+- Scheduling information (heavy rail only) ("trainsheets")
+  - Which trip the train is on. These are internal IDs, and not GTFS trips.
+  - What the subsequent trip is.
+  - Terminal departure times of trips, with dispatcher-entered "offsets" in minutes
+- "Device" information, like state of switches and terminal bells, and hold lights
+  - We mostly don't use any of this.
+
+RTR uses this raw data to keep track of where the known subway vehicles are. It attempts to match
+current revenue OCS trips to a plausible GTFS trip, creating an "ADDED" trip if none match. It
+publishes this data as VehiclePositions.
+
+RTR also generates predictions for most revenue vehicles' current trips and some subsequent trips.
+It publishes this data as TripUpdates. Unmatched GTFS trips it publishes as CANCELED.
+
+In the future, we may get raw GPS updates from all light rail vehicles once every 5 seconds, both
+in and out of revenue service.
+
+The RTR data is GTFS-rt-esque; The TripUpdates also includes our "boarding_status" extension for,
+e.g., "Stopped 4 stops away" for Realtime Signs, a passthrough_time field for non-revenue trains
+for Realtime Signs to play the "please stand back from the yellow line" message, route_pattern_id
+for ADDED trips. The VehiclePositions include "consist" information.
+
+Realtime Signs consumes the RTR `_enhanced.json` directly, rather than via the public API for
+those extra fields above. Glides consumes it directly, and would ideally like the VehiclePositions
+of all light rail trains, in service or not.
+
+### With CentralStation
+
+RTR will change to behave more like how we want Swiftly to behave.
+
+It will consume VehiclePositions data from CentralStation, and importantly, those vehicle
+positions will have a GTFS trip tagged. This will simplify RTR as a large part of the message
+handling logic is about deciding what trip a vehicle is on.
+
+It will also consume the ServiceChanges feed, which is how CentralStation will convey the TSCH
+offsets it receives.
+
+As a bonus, by relying on GTFS trips and ServiceChanges, RTR won't have to rely on handcrafted
+patterns, and should be able to predict, e.g., new stations or closed stations automatically.
```

Comment:
Yeah, I'd think RTR needs to have patterns at the track circuit level, which is not available from GTFS.

Reply:
@skyqrose is there any chance that Glides users might want a track circuit level view of the Green Line (for the areas where that exists, at least?). That seems like it might be relevant as well.

Reply:
Yes definitely. And though the track layout is static/schedule data, the assignment of trains to those track circuits in real time definitely belongs to this RFC, and would be great to have in any internal realtime data feed.